index.js

```diff
@@ -35,7 +35,7 @@ module.exports = function (app, options) {
     var names = fs.readdirSync(localeDir);
     for (var i = 0; i < names.length; i++) {
       var name = names[i];
-      if (!/\.(js|json|properties)$/.test(name)) {
+      if (!/\.(?:js|json|properties)$/.test(name)) {
```

Comment:
NO. catch group will catch it, whether `test` or `exec`, catch group can be used in regexp self, like `/(["']).*?\1/`. `\1` is reference the catch group.